### DISCUSSION TITLE
Pointless "Success!" message

### DISCUSSION THREAD

**Discussion by Madis0** [MarcusOtter/discord-needle#74] [General]
**Describe the bug**
Closing a thread results in a private "Success!" message, despite there being a public archival message right next to it.

**To Reproduce**
1. Post in a channel with auto-threading
2. Archive the created thread
3. See the issue

**Expected behavior**
Only see the public message
> Thread was archived by...

**Screenshots**
![Screenshot_20220204_081602](https://user-images.githubusercontent.com/8611110/152481569-1d54de96-1d3d-4b13-99ca-84419dd6d66d.png)

**System specifications (please complete the following information if applicable):**
N/A

**Additional context**

**Comment by MarcusOtter** [comment]
I really dislike this message too. I would love if someone else could chime in on how to resolve this, because I don't really know how. If we don't reply to the interaction, the user will get a "this interaction failed" message. And I don't want the public message to reply to the interaction, as that adds a lot of visual bloat:

![bild](https://user-images.githubusercontent.com/35617441/152509422-f2bc8978-f069-4d23-855d-64710f5062bf.png)

**Comment by Madis0** [comment]
> And I don't want the public message to reply to the interaction, as that adds a lot of visual bloat

I would say it is still a lot better because it is just a reply indicator, not a separate message that attracts more attention on its own.

**Comment by MarcusOtter** [comment]
The way it is now, it only adds visual bloat to the user that closes the thread (visual bloat that will disappear once the client is restarted, and it can be removed more easily by simply clicking the "dismiss" button).

The alternative adds noise for everyone permanently. I'm not sure I prefer how it was before

**Comment by Madis0** [comment]
Then, how about going one step further and letting servers configure whether this "thread was archived..." message is displayed to everyone or just the user? 
Where everyone would be a reply and just the user would be a similar message to the current "success!".

That way the bloat could be reduced even further - remember, Discord itself doesn't announce when a thread is archived or timer adjusted 😛

**Comment by MarcusOtter** [comment]
Discord doesn't do a lot of things, auto-threading for example 😉 I don't think Discord makes it clear enough that a thread is archived as it is now, and it's good to know for everyone to know who decided to archive the thread (was it the user that considered their thread solved? was it a moderator that thought it was too off-topic?).

The best thing would be if Discord would just show in a banner when the thread was archived and by who. I think configuring whether "thread was archived.." message is displayed would fall under the verbosity setting we will have later.

I don't see any good options right now but I could be missing something. Perhaps we don't have to reply to an interaction to make it "succeed" for the user?

**Comment by MarcusOtter** [comment]
One very ugly workaround I thought about now is to reply to the interaction publicly, immediately delete the message, then post the message we want (which is not a reply). We need to post the first reply publicly since bots cannot delete ephemeral (hidden) messages.

**Comment by n1ckoates** [comment]
This issue should probably be converted to a discussion.

**Comment by n1ckoates** [comment]
We could edit the initial message that has the button, just adding a space or something. This counts as a response as far as Discord is concerned.

**Reply by MarcusOtter** [reply]
That's a fantastic idea, it's so hacky but so good.

**Comment by n1ckoates** [comment]
This should be resolved by #90.

**Comment by MarcusOtter** [comment]
Resolved by #90, will go out to the hosted bot soon!
